### PR TITLE
HDDS-4417. Simplify Ozone client code with configuration object

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

In [HDDS-4185 we agreed](https://github.com/apache/ozone/pull/1374#issuecomment-696169868)  to introduce a new configuration for the Ozone client to adjust the behavior of incremental byte buffer.

When I started to add it I realized the code is already very complex as all the required configuration values are propagated manually to the Key/Block/ChunkOutputstream as induvidual constructor parameters.

In this patch I simplify the structure with using only one POJO with configuration annotations.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4417

## How was this patch tested?

With full CI on my fork.

### CODE REVIEW

**Comment by bshashikant** [comment]
@elek , can u check the failed checks. The changes in general look good to me.

**Comment by elek** [comment]
> @elek , can u check the failed checks. The changes in general look good to me. Will look into it in more detail.

Thanks to check it. Yes, I am fixing the failures. It was green on my fork, so must be some silly mistake.

**Review by bshashikant — APPROVED** [review]

**Comment by bshashikant** [comment]
Thanks @elek for the contribution.

**Review by adoroszlai — COMMENTED** [review]
Thanks @elek for this refactoring.  I noticed a few issues, can they be addressed in an addendum?

**Comment by elek** [comment]
Thanks the suggestions @adoroszlai Will create a new PR with the same HDDS-4417 tag shortly.

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClient.java [hunk 1]
```diff
@@ -49,6 +51,7 @@
   @BeforeClass
   public static void init() throws Exception {
     OzoneConfiguration conf = new OzoneConfiguration();
+    conf.setFromObject(new OzoneClientConfig());
```
[the remark this answers is not in the document]
Comment (elek) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestOzoneRpcClient.java:54]: Yes, it's supposed to be set. I was confused by the other pattern used elsewhere:

 1. create OzoneConfig
 2. set custom properties by setter
 3. Here we need to export OzoneClientConfig
 4. modify tags
 5. reset OzoneClientConfig to OzoneConfig

**Comment by adoroszlai** [comment]
> Will create a new PR with the same [HDDS-4417](https://issues.apache.org/jira/browse/HDDS-4417) tag shortly.

Done: https://github.com/apache/ozone/pull/1581

**Comment by elek** [comment]
> Done: #1581

Oh, missed it. I also did it. But let me merge this one. Thanks for the fixes.

**Comment by adoroszlai** [comment]
> > Done: #1581
> 
> Oh, missed it. Thanks for the fixes.

It's not yet in, needs review.

**Pull request merged**